### ISSUE TITLE
Notch Plugin gives inaccurate value

### ISSUE THREAD

**Issue by Tarek200** [Scirra/Construct-3-bugs#3431]
Notch Plugin gives inaccurate values, this makes it hard to detect on which side is the Notch located to adjust the UI

## Attach a .c3p

[.C3p](https://www.dropbox.com/s/9yorf5f03f8qck0/notch%20c3%20bug1.c3p?dl=0)

## Steps to reproduce

1-Pick up an iPhone that has Notch and change screen orientation to Landscape mode and Notch should be situated at the Left side
[Like This Picture:](https://www.dropbox.com/s/5xsqfew3ocw8rdo/scirra.png?dl=0)

2-Run the .c3p

## Observed result

Wrong value for the expression:
PlatformInfo.SafeAreaInsetRight = 44

## Expected result

Value it should PlatformInfo.SafeAreaInsetRight = 0 
as there is not Notch on the right side

## More details

-Tested on Preview & Exported with Cordova
-Tested Device iPhone Xs Max
-C3 Version: r172 Beta

## System details

<details><summary>View details</summary>

Platform information
Browser: Chrome
Browser version: 77.0.3865.120
Browser engine: Chromium
Browser architecture: 64-bit
Context: browser
Operating system: Windows
Operating system version: 10
Operating system architecture: 64-bit
Device type: desktop
Device pixel ratio: 1
Logical CPU cores: 8
Approx. device memory: 8 GB
User agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.120 Safari/537.36
C3 release: r172 (beta)
Language setting: en-US

Local storage
Storage quota (approx): 19 gb
Storage usage (approx): 586 mb (3%)
Persistant storage: Yes

Browser support notes
This list contains missing features that are not required, but could improve performance or user experience if supported.

Nothing is missing. Everything is OK!
WebGL information
Version string: WebGL 2.0 (OpenGL ES 3.0 Chromium)
Numeric version: 2
Supports NPOT textures: yes
Supports GPU profiling: yes
Supports highp precision: yes
Vendor: Google Inc.
Renderer: ANGLE (NVIDIA GeForce GTX 1070 Direct3D11 vs_5_0 ps_5_0)
Major performance caveat: no
Maximum texture size: 16384
Point size range: 1 to 1024
Extensions:

EXT_color_buffer_float
EXT_disjoint_timer_query_webgl2
EXT_float_blend
EXT_texture_filter_anisotropic
KHR_parallel_shader_compile
OES_texture_float_linear
WEBGL_compressed_texture_s3tc
WEBGL_compressed_texture_s3tc_srgb
WEBGL_debug_renderer_info
WEBGL_debug_shaders
WEBGL_lose_context
Audio information
System sample rate: 48000 Hz
Output channels: 2
Output interpretation: speakers
Supported decode formats:

Ogg Opus (audio/ogg; codecs=opus)
WebM Opus (audio/webm; codecs=opus)
Ogg Vorbis (audio/ogg; codecs=vorbis)
WebM Vorbis (audio/webm; codecs=vorbis)
MPEG-4 AAC (audio/mp4; codecs=mp4a.40.5)
MP3 (audio/mpeg)
FLAC (audio/flac)
PCM WAV (audio/wav; codecs=1)
Supported encode formats:

WebM Opus (audio/webm; codecs=opus)
Video information
Supported decode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)
Ogg Theora (video/ogg; codecs=theora)
H.264 (video/mp4; codecs=avc1.42E01E)
Supported encode formats:

WebM VP9 (video/webm; codecs=vp9)
WebM VP8 (video/webm; codecs=vp8)

</details>

**Comment by Tarek200** [comment]
Also if you can make sure to count for the situation when the user Rotate the iPhone at RunTime, it should update the Values 
Example:
If Notch is situated at the Left side:
Values should be:
PlatformInfo.SafeAreaInsetLeft = 44
PlatformInfo.SafeAreaInsetRight = 0

Then if you Rotate the iPhone at the RunTime positioning the Notch at the Right side:
Values should update to:
PlatformInfo.SafeAreaInsetLeft = 0
PlatformInfo.SafeAreaInsetRight = 44

**Comment by shortercode** [comment]
I think I should note that the cordova "notch" plugin that we use only exists on Android, and as far as I'm aware has no interaction with the safe area values.

Also the iPhone X has 2 safe zones, one on the bottom and one at the top. So it's expected for both to be non-zero.

**Comment by AshleyScirra** [comment]
If you get nonzero values, the plugin is simply reporting the browser-provided [safe-area-inset-*](https://developer.mozilla.org/en-US/docs/Web/CSS/env) values. I would assume the browser is reporting these values correctly (and if they aren't correct, it would need to be reported to the browser vendor). In particular the safe area does *not* actually specify where a notch is: it only defines a safe rectangular area inside the screen where all pixels are visible and not taken up by a notch. This also includes rounded corners, and note all four corners of the screen are rounded, so all four corners need to be moved inwards a little to avoid the corners. Additionally making the margins symmetrical may be a conscious design choice to give a centered safe area. Also note we update these values when the screen size changes (which includes when the orientation changes).

**Comment by Tarek200** [comment]
**Thanks for the all the info:**

I'm really stuck because I need to find a way to detect on which Area is the Node as I need to adjust the UI, My game is locked to Landscape mode.
Here some pictures showing the UI,  the UI changes Positions according to phone Orientation

[Notch at Left:](https://www.dropbox.com/s/d5gn0h13wdopqvg/left.png?dl=0)
[Notch at Right:](https://www.dropbox.com/s/bmnmzz9ucglvv1b/right.png?dl=0)



Right now with the official Notch Plugin, we can detect only if the phone has a Notch which is still really helpful thanks for adding it.

But without being able to detect on which area is the Notch the plugin will not be really helpful for many Games like mine.

**It could all be resolved if you could add a way to Detect "Phone Orientation"**

[Phone Orientation](https://www.dropbox.com/s/xyr9el7vbn2pb4c/iphone-orientation.png?dl=0)


**Detect "Phone Orientation" +  Notch Plugin will give us the combo that will solve all the problems for UI**

Will it be possible that you include  Detect "Phone Orientation"? 
we really need this because other ways many UI games made c3 is gonna be a disaster like mine.
Thanks for your Time

**Comment by AshleyScirra** [comment]
Your "Notch at right" picture shows the HUD outside the safe area. The screen corners could cut off the image.

You can already detect the orientation too.

Devices could in theory have multiple notches too, so you just have to stick to the safe area to know for sure that the content is going to be visible.

**Comment by Tarek200** [comment]
I'm developing for ios and I haven't seen any iPhone with double Notch yet. 
Also, I had a quick search on Android and I could find only one phone with double Notch maybe are more but I couldn't see any, even if they are a few more it's like 99% of the phones haven't got double Notch.
Many apps like mine they change the UI according to the Device Orientation and they haven't got any problem with the safe Area as long you know where the Notch is located.
-----------

> You can already detect the orientation too.

Sorry I tried all possible expressions from (PlatformInfo & Browser) plugins and I couldn't detect the Phone orientation Direction, I could only detect if the if Phone is on (Landscape or Portrait) mode which leaves us with the same problem where is the Notch located?

Can you please explain to me how do you get the Device orientation but giving the Orientation on Values: 
**Example:**
**Device Rotated to the Left = 90
Device Rotated to the Right = -90
Device No Rotated = 0
Device Rotated to the Bottom = 180**

**Like they explain here:** 
[DETECT IOS DEVICE ORIENTATION WITH JAVASCRIPT](http://www.williammalone.com/articles/html5-javascript-ios-orientation/)




Its really important feature can you please add it to the Plugin, I cannot just use the safe Area to positioned the UI as if the phone is rotated to the Right the UI it will be nearly on the middle of the screen blocking the view for no reason when has plenty space behind it, it will look awkward as it will have plenty space behind the UI and it will look like just floating in the middle for no reason, I haven't seen any game doing this, they are going to have fun of us lol.

**Comment by AshleyScirra** [comment]
The precise device orientation still doesn't tell you where a notch is. I'm not aware of any web platform feature that provides that information at the moment, I think the safe area is all we get.

**Comment by Tarek200** [comment]
With the help of the precise device orientation, we can detect where is the Notch located 

**Example:**
All the iOs Devices that have Notch they have it on the Top if the phone is not Rotated, so that will help with the starting point like this:

**Precise device orientation = 0**  >>>Notch at the Top
**Precise device orientation = -90**  >>>Notch at the Right
**Precise device orientation = 180** >>>Notch at the Bottom
**Precise device orientation = 90** >>>Notch at the Left

**Here is a visual example:**
[Notch Detector: ](https://www.dropbox.com/s/7cychlszm06rfgc/Device%20Orientations%20.png?dl=0)


1-So the idea is to use the (Notch plugin) just to detect if the phone has a Notch 

2-Then if yes use the "Precise device orientation" to detect where is the Notch located

I don't see any reason why it wouldn't work?

**Comment by shortercode** [comment]
@Tarek200 if your aiming specifically to support the iOS notched devices then as far as I'm aware your correct that they only have a notch at the top. Although the horizontal grey bar at the bottom of the display which represents the device menu gesture has it's own safe area, and you should avoid placing anything under that.

Knowing the size of the notch and it's position relative in a given orientation yes you can lay it out according to your example. I expect the 3rd safe area in landscape is to encourage application to follow a symmetrical layout on device. You might be able to use a pinched style layout, where it's narrower in the middle than the top and bottom. I've seen a few demos do this.

**Comment by Tarek200** [comment]
**Hi shortercode ho you doing**

I understand what you guys saying but the truth is that there are two problems I need to solve:

**1-Is to place the UI as I explained above
2-I need to cover the Notch with a fake black bar**

Say for problem 1 I follow your advice and just place the UI on the safe area ok no problem but I will be still stuck on problem 2 which I will still need to detect where is the Notch located to add a fake black background to the Notch area.
I didn't want to talk about the two problems at the same time so I don't confuse the thread.

**Here is an example of adding a fake black background to the Notch:**

**Normal no modified:**
![without-bar](https://user-images.githubusercontent.com/41597700/67505706-a885cb00-f683-11e9-94ba-611605a7fde4.png)


**But I would like to add a fake black bar like this:**
![bar2](https://user-images.githubusercontent.com/41597700/67505756-c0f5e580-f683-11e9-9b0a-3b30ac845903.png)

Just to make it look more natural and symmetric as when I place the UI looks weird having the Notch with the two empty spaces there, thats why I need to cover it.



If you no know any other solution to add the fake black background to the Notch area I'm happy to do it, I opened a thread about a month ago but no one replied to it.

[Thread Link:](https://www.construct.net/en/forum/construct-3/general-discussion-7/possible-cover-iphone-xs-xs-147098)


Thanks for your time
